bigquery/src/main/java/org/apache/zeppelin/bigquery/BigQueryRestrictedAccessInterpreter.java

```diff
@@ -0,0 +1,299 @@
+/*
+* Copyright 2016 Google Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package org.apache.zeppelin.bigquery;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.*;
+
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.auth.oauth2.ServiceAccountCredentials;
+import com.google.cloud.bigquery.*;
+import com.google.cloud.http.HttpTransportOptions;
+import org.apache.zeppelin.interpreter.Interpreter;
+import org.apache.zeppelin.interpreter.InterpreterContext;
+import org.apache.zeppelin.interpreter.InterpreterResult;
+import org.apache.zeppelin.interpreter.thrift.InterpreterCompletion;
+import org.apache.zeppelin.scheduler.Scheduler;
+import org.apache.zeppelin.scheduler.SchedulerFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+
+/**
+ * BigQuery interpreter for Zeppelin.
+ * <p>
+ * <ul>
+ * <li>{@code zeppelin.bigquery.project_id} - Project ID in GCP</li>
+ * <li>{@code zeppelin.bigquery.wait_time} - Query Timeout in ms</li>
+ * <li>{@code zeppelin.bigquery.max_no_of_rows} - Max Result size</li>
+ * </ul>
+ * <p>
+ * <p>
+ * How to use: <br/>
+ * {@code %bigquery.sql<br/>
+ * {@code
+ * SELECT departure_airport,count(case when departure_delay>0 then 1 else 0 end) as no_of_delays
+ * FROM [bigquery-samples:airline_ontime_data.flights]
+ * group by departure_airport
+ * order by 2 desc
+ * limit 10
+ * }
+ * </p>
+ */
+
+public class BigQueryRestrictedAccessInterpreter extends Interpreter {
```

Comment:
Can we not extend the other BigQuery interpreter instead of having to re-implement everything?